x-pack/plugins/reporting/server/config/create_config.ts

```diff
@@ -33,21 +33,25 @@ export function createConfig(
     encryptionKey = crypto.randomBytes(16).toString('hex');
   }
 
-  const { kibanaServer: reportingServer } = config;
+  const { kibanaServer: reportingServer, statefulSettings } = config;
   const serverInfo = core.http.getServerInfo();
   // set kibanaServer.hostname, default to server.host, don't allow "0.0.0.0" as it breaks in Windows
   let kibanaServerHostname = reportingServer.hostname
     ? reportingServer.hostname
     : serverInfo.hostname;
 
-  if (
+  if (statefulSettings.enabled === false) {
+    logger.info(
+      `Found 'server.host: "0.0.0.0"' in Kibana configuration. Overriding 'xpack.reporting.kibanaServer.hostname' with 'localhost'`
+    );
+  } else if (
```

Comment:
This should be back to what it was, and not pair the kibanaServer concern with "statefulSettings"
```suggestion
if (
```

I think this will mean the unit test changes will be reverted too.